crawler.py

```diff
@@ -333,6 +339,73 @@ def __crawl(self, current_url):
 
 			self.urls_to_crawl.add(link)
 
+	def write_sitemap_output(self):
+		are_multiple_sitemap_files_required = \
+			len(self.url_strings_to_output) > self.MAX_URLS_PER_SITEMAP
+
+		# When there are more than 50,000 URLs, the sitemap specification says we have
+		# to split the sitemap into multiple files using an index file that points to the
+		# location of each sitemap file.  For now, we require the caller to explicitly
+		# specify they want to create an index, even if there are more than 50,000 URLs,
+		# to maintain backward compatibility.
+		#
+		# See specification here:
+		# https://support.google.com/webmasters/answer/183668?hl=en
+		if are_multiple_sitemap_files_required and self.as_index:
+			self.write_index_and_sitemap_files()
+		else:
+			self.write_single_sitemap()
+
+	def write_single_sitemap(self):
+		self.write_sitemap_file(self.output_file, self.url_strings_to_output)
+
+	def write_index_and_sitemap_files(self):
+		sitemap_index_filename, sitemap_index_extension = os.path.splitext(self.output)
+
+		num_sitemap_files = math.ceil(len(self.url_strings_to_output) / self.MAX_URLS_PER_SITEMAP)
+		sitemap_filenames = []
+		for i in range(0, num_sitemap_files):
```

Comment:
It probably doesn't matter, but I notice that in the [Google documentation](https://support.google.com/webmasters/answer/75712?visit_id=637359494875837645-2439915332&rd=1), they 1-index the sitemaps whereas they're 0-indexed there.  Curious to hear your thoughts there.

Reply:
Its really matter ?

Reply:
Yeah, 99% sure it doesn't matter, so either way is fine I reckon.